src/test/java/com/snowplowanalytics/snowplow/tracker/TrackerTest.java

```diff
@@ -75,6 +65,51 @@ public void setUp() {
 
     // --- Event Tests
 
+    @Test
+    public void testTrackReturnsEventIdIfSuccessful() throws InterruptedException {
+        // a list to allow for eCommerceTransaction
+        List<String> result = tracker.track(Unstructured.builder()
+                .eventData(new SelfDescribingJson(
+                        "iglu:com.snowplowanalytics.snowplow/example/jsonschema/1-0-0",
+                        ImmutableMap.of("foo", "bar")
+                ))
+                .build());
+
+        Thread.sleep(500);
+
+        // this throws an exception if it's not a valid UUID string
+        UUID.fromString(result.get(0));
```

Comment:
I think we can make this explicit in the test.
Probably the result would be the same but raising an exception in the tests could have an unexpected behaviour not just like a failure. I can't find any reference to support this and it could be I remember cases related to another platform/language (ObjC ?!) but JUnit offers some assert methods for this so it can be worth to use them: https://www.baeldung.com/junit-assert-exception

Reply:
Raising an exception does just fail the test, but yeah it would be nicer to assert it explicitly

Reply:
No need to change this, I misunderstood the test. The comment made me think to an exception testing but actually you are testing it's successful 👍

Reply:
I've changed it anyway. You misread it so it could have been clearer :)